mypy/checker.py

```diff
@@ -1240,7 +1255,7 @@ def check_func_def(
                     # have no good way of doing this.
                     #
                     # TODO: Find a way of working around this limitation
-                    if len(expanded) >= 2:
+                    if len(expanded) >= 2 or self._is_empty_generator(item):
```

Comment:
Maybe update the multiline comment immediately above this line? We're not just suppressing reachability warnings for TypeVars with value restrictions anymore

Reply:
Yeah, I didn't want to muddy the blame for that comment 🤣

Reply:
e31efd3a07d08af89715190484e77bdc1d7f2614